opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -46,95 +59,162 @@ def __init__(
         self._instrumentation_info = instrumentation_info
         self._meter_provider = meter_provider
 
-    def create_counter(self, name, unit=None, description=None) -> Counter:
-        # FIXME implement this method
-        pass
+    def create_counter(self, name, unit=None, description=None) -> APICounter:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_counter(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_up_down_counter(
         self, name, unit=None, description=None
-    ) -> UpDownCounter:
-        # FIXME implement this method
-        pass
+    ) -> APIUpDownCounter:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_up_down_counter(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_observable_counter(
         self, name, callback, unit=None, description=None
-    ) -> ObservableCounter:
-        # FIXME implement this method
-        pass
+    ) -> APIObservableCounter:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_observable_counter(
+            self._instrumentation_info, name, unit, description
+        )
 
-    def create_histogram(self, name, unit=None, description=None) -> Histogram:
-        # FIXME implement this method
-        pass
+    def create_histogram(
+        self, name, unit=None, description=None
+    ) -> APIHistogram:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_histogram(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_observable_gauge(
         self, name, callback, unit=None, description=None
-    ) -> ObservableGauge:
-        # FIXME implement this method
-        pass
+    ) -> APIObservableGauge:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_observable_gauge(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_observable_up_down_counter(
         self, name, callback, unit=None, description=None
-    ) -> ObservableUpDownCounter:
-        # FIXME implement this method
-        pass
+    ) -> APIObservableUpDownCounter:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_observable_up_down_counter(
+            self._instrumentation_info, name, unit, description
+        )
 
 
 class MeterProvider(APIMeterProvider):
     """See `opentelemetry._metrics.MeterProvider`."""
 
     def __init__(
         self,
+        metric_exporters: Sequence[MetricExporter] = (),
```

Comment:
I don't believe the `MeterProvider` should be responsible for the exporters. See [this](https://github.com/open-telemetry/opentelemetry-python/pull/2280/files#r763141002).

Reply:
Maybe :shrug:, but the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#meter-creation) says this:

> Configuration (i.e., MetricExporters, MetricReaders and Views) MUST be managed solely by the MeterProvider and the SDK MUST provide a way to configure all options that are implemented by the SDK. This MAY be done at the time of MeterProvider creation if appropriate.

Reply:
That is correct. The configuration can be done in the `MeterProvider` but it should be passed down through the `MetricReader`. The exporters can have a reference to the corresponding `MetricReader`.

Reply:
What about [`MeterProvider.ForceFlush`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#forceflush)? The `MeterProvider` needs to know which `MetricReader` and `MetricExporter` instances it has so it can notify them:

> This method provides a way for provider to notify the registered MetricReader and MetricExporter instances, so they can do as much as they could to consume or send the metrics.

Reply:
If there is a logic path from the `MeterProvider` to the exporter, then I don't believe `MeterProvider` needs to have the exporters stored in memory. `MeterProvider` -> `MetricReader` -> `Exporter`. Similar to `TracerProvider` -> `SpanProcessor` -> `Exporter`.

Reply:
^ This could be more inconvenient then I am making it seem because `MetricReader`, `MeasurementProcessor` aren't defined yet so maybe we can just leave this for a different PR.

Reply:
There's a little bit of discussion of this here: https://github.com/open-telemetry/opentelemetry-specification/issues/2129#issuecomment-976180751

I think the specification wording is a little misleading here. If a MetricExporter is always paired with the reader, and the reader flushes the exporter, I think it meets the requirement "provides a way for provider to notify..."

Reply:
Marking this conversation as unresolved, solving this is being tracked in #2316